crypto/s2n_aead_cipher_chacha20_poly1305.c

```diff
@@ -105,6 +111,8 @@ static int s2n_aead_chacha20_poly1305_decrypt(struct s2n_session_key *key, struc
 
     S2N_ERROR_IF(evp_decrypt_rc != 1, S2N_ERR_DECRYPT);
 
+    /* While we verify the content of out_len in s2n_aead_chacha20_poly1305_encrypt, we refrain from this here. This is to avoid doing any branching before the ciphertext is verified. */
+
```

Comment:
Does that not happen on line 110 and 112?

Reply:
Line 107 decrypts and line 110 verifies the tag, true. But, there is no branching based on the return value between the two calls before after authentication. This happens in line 112.

In OpenSSL you must authenticate via a second API call and, technically, does allow release of unauthenticated data for AE ciphers, but it is a red herring. S2N protect against this by making the process atomic through a call to `s2n_aead_chacha20_poly1305_decrypt`.

I could check `*out_len` is `0` after line 112.